python/dialects/_AIE_syntax.py

```diff
@@ -0,0 +1,257 @@
+from ._AIE_ops_gen import *
+from ._AIE_util import *
+from .._mlir_libs._aieMlir import register_dialect, ObjectFifoType, ObjectFifoSubviewType
+
+
+# Helper function that returns the index of a named objectFifo port.
+def objectFifoPortToIndex(port):
+    port_index = -1
+    if (port == "Produce"):
+        port_index = 0
+    elif (port == "Consume"):
+        port_index = 1
+    else:
+        port_index = -1
+    return port_index
+
+
+# Helper function that returns the index of a named WireBundle.
+def wireBundleToIndex(port):
+    port_index = -1
+    if (port == "Core"):
+        port_index = 0
+    elif (port == "DMA"):
+        port_index = 1
+    elif (port == "FIFO"):
+        port_index = 2
+    elif (port == "South"):
+        port_index = 3
+    elif (port == "West"):
+        port_index = 4
+    elif (port == "North"):
+        port_index = 5
+    elif (port == "East"):
+        port_index = 6
+    elif (port == "PLIO"):
+        port_index = 7
+    elif (port == "NOC"):
+        port_index = 8
+    elif (port == "Trace"):
+        port_index = 9
+    else:
+        port_index = -1
+    return port_index
+
+
+def dim_tuple_attr_builder(wrap, stepsize):
+    return Attribute.parse(f"#AIE.DimTuple<{wrap}, {stepsize}>")
+
+
+@register_attribute_builder("AIE_DimTupleArrayAttr")
+def dim_tuple_array_attr_builder(tups: List[tuple], context=None):
+    tups = list(map(lambda t: dim_tuple_attr_builder(*t), tups))
+    return Attribute.parse(
+        f'#AIE<DimTupleArray[{", ".join(map(str, tups))}]>', context=context
+    )
+
+
+@register_attribute_builder("AIE_DimTupleArrayArrayAttr")
+def dim_tuple_array_array_attr_builder(tup_arrs: List[List[tuple]], context=None):
+    tup_arrs = list(map(dim_tuple_array_attr_builder, tup_arrs))
+    return Attribute.parse(
+        f'#AIE<DimTupleArrayArray[{", ".join(map(str, tup_arrs))}]>', context=context
+    )
+
+
+#### AIE Wrappers ####
+
+
+# Create and print ModuleOp.
+def constructAndPrintInModule(f):
+    with Context() as ctx, Location.unknown():
+        register_dialect(ctx)
+        module = Module.create()
+        with InsertionPoint(module.body):
+            f()
+        print(module)
+
+
+# Create an aie device on specified target architecture.
+class Device(DeviceOp):
```

Comment:
This is going to break as soon as we bump due to this PR I landed upstream https://github.com/llvm/llvm-project/pull/68853.

Reply:
Hey Maks, thank you for your detailed review! I looked through the PRs you mentioned above. Could you please help me understand why this would break exactly? In the PR above it looks like extensions that were calling super.__init__() can stay unchanged.

Reply:
Sorry I didn't realize that `AIEDevice` is already an `EnumAttr`. So you can get this for free by signing up [using `GEN_ENUM_BINDINGS`](https://github.com/llvm/llvm-project/blob/main/mlir/python/CMakeLists.txt#L65).